alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -100,19 +100,34 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
         return clusterState.routingTable.hasIndex(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
     }
 
-    /**
-     * From given index mapping node, extracts fieldName -> fieldProperties pair
-     */
-    fun extractField(node: MutableMap<String, Any>, currentPath: String): Pair<String, MutableMap<String, Any>> {
+    fun traverseMappingsAndUpdate(
```

Comment:
Can you add comments in the different logic flows of this function? It is kind of hard to follow what it is doing.

Reply:
done